src/panels/lovelace/entity-rows/hui-media-player-entity-row.js

```diff
@@ -103,22 +103,27 @@ class HuiMediaPlayerEntityRow extends LocalizeMixin(PolymerElement) {
   _computeMediaTitle(stateObj) {
     if (!stateObj || this._isOff(stateObj.state)) return null;
 
+    let prefix;
+    let suffix;
+
     switch (stateObj.attributes.media_content_type) {
       case "music":
-        return `${stateObj.attributes.media_artist}: ${
-          stateObj.attributes.media_title
-        }`;
+        prefix = stateObj.attributes.media_artist;
+        suffix = stateObj.attributes.media_title;
+        break;
       case "tvshow":
-        return `${stateObj.attributes.media_series_title}: ${
-          stateObj.attributes.media_title
-        }`;
+        prefix = stateObj.attributes.media_series_title;
+        suffix = stateObj.attributes.media_title;
+        break;
       default:
-        return (
+        prefix =
           stateObj.attributes.media_title ||
           stateObj.attributes.app_name ||
-          stateObj.state
-        );
+          stateObj.state;
+        suffix = "";
     }
+
+    return `${prefix || ""}${prefix && suffix ? ": " : ""}${suffix || ""}`;
```

Comment:
So it was a great cleanup until I saw this line… this is kinda wild.

I think that instead of `prefix` and `suffix`, we just have a list `parts` that we push to:

```ts
const parts = [];

parts.push(stateObj.attributes.media_title);

return parts.filter(Boolean).join(": ");
```

Reply:
Or if we want to keep it very readable, just a simple if…else if…else

```js
if (prefix && suffix) {
  return `${prefix}: ${suffix}`;
} else if (prefix || suffix) {
  return prefix || suffix;
} 
return "";
```

Reply:
That's pretty much what I had originally @balloob, although a bit more condensed (using `else if` instead of just multiple `if`s). What do you think of this even more condensed version? I still thing it's pretty easy to understand what it's doing.

```JavaScript
return prefix && suffix ? `${prefix}: ${suffix}` : (prefix || suffix || "");
```

Reply:
I like that 👍

Reply:
Changed :+1: